catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/cache/solr/impl/ValidationQueryFactory.java

```diff
@@ -41,25 +44,30 @@ public ValidationQueryFactory(FilterAdapter filterAdapter, FilterBuilder filterB
         builder = filterBuilder;
     }
 
-    public QueryRequest getQueryRequestWithValidationFilter(QueryRequest input,
-            boolean showInvalidMetacards) {
+    QueryRequest getQueryRequestWithValidationFilter(QueryRequest input, Boolean showErrors,
+            Boolean showWarnings) {
         Query inputQuery = input.getQuery();
         try {
-            if (showInvalidMetacards || adapter.adapt(input.getQuery(),
+            if ((showErrors && showWarnings) || adapter.adapt(input.getQuery(),
```

Comment:
Why `&&` here? Shouldn't errors and warnings be adapted separately, since they can be shown separately?